src/ansys/aedt/core/extensions/hfss/move_it.py

```diff
@@ -22,248 +22,180 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-# Extension template to help get started
-
+from dataclasses import dataclass
+import os
 from pathlib import Path
-from tkinter import messagebox
+import tkinter
+from tkinter import ttk
 
 import numpy as np
 from scipy.interpolate import CubicSpline
 
 import ansys.aedt.core
 from ansys.aedt.core import get_pyaedt_app
+import ansys.aedt.core.extensions
+from ansys.aedt.core.extensions.misc import ExtensionCommon
+from ansys.aedt.core.extensions.misc import ExtensionCommonData
 from ansys.aedt.core.extensions.misc import get_aedt_version
 from ansys.aedt.core.extensions.misc import get_arguments
 from ansys.aedt.core.extensions.misc import get_port
 from ansys.aedt.core.extensions.misc import get_process_id
 from ansys.aedt.core.extensions.misc import is_student
 from ansys.aedt.core.internal.errors import AEDTRuntimeError
 
-port = get_port()
-version = get_aedt_version()
-aedt_process_id = get_process_id()
-is_student = is_student()
+PORT = get_port()
+VERSION = get_aedt_version()
+AEDT_PROCESS_ID = get_process_id()
+IS_STUDENT = is_student()
 
 # Extension batch arguments
-extension_arguments = {"choice": "", "velocity": 1.4, "acceleration": 0.0, "delay": 0.0}
-extension_description = "Move It"
-
-
-def frontend():  # pragma: no cover
-    """Create the GUI for the extension."""
-    import tkinter
-    import tkinter.ttk as ttk
-
-    import PIL.Image
-    import PIL.ImageTk
-
-    from ansys.aedt.core.extensions.misc import ExtensionTheme
-
-    app = ansys.aedt.core.Desktop(
-        new_desktop=False,
-        version=version,
-        port=port,
-        aedt_process_id=aedt_process_id,
-        student_version=is_student,
-    )
-
-    active_project = app.active_project()
-
-    if not active_project:  # pragma: no cover
-        app.logger.error("No active project.")
-
-    active_design = app.active_design()
-
-    if not active_design:  # pragma: no cover
-        app.logger.error("No active design.")
-
-    project_name = active_project.GetName()
-    design_name = active_design.GetName()
-
-    hfss = get_pyaedt_app(project_name, design_name)
-
-    if hfss.design_type != "HFSS":  # pragma: no cover
-        app.logger.error("Active design is not HFSS.")
-        hfss.release_desktop(False, False)
-        output_dict = {"choice": "", "file_path": ""}
-        return output_dict
-
-    # Create UI
-    master = tkinter.Tk()
+EXTENSION_DEFAULT_ARGUMENTS = {"choice": "", "velocity": 1.4, "acceleration": 0.0, "delay": 0.0}
+EXTENSION_TITLE = "Move It"
 
-    master.geometry()
 
-    master.title(extension_description)
+@dataclass
+class MoveItExtensionData(ExtensionCommonData):
+    """Data class containing user input and computed data."""
 
-    # Detect if user close the UI
-    master.flag = False
+    choice: str = ""
+    velocity: float = 0.0
```

Comment:
```suggestion
    velocity: float = 1.4
```
I would align with the default argument values. Maybe using `EXTENSION_DEFAULT_ARGUMENTS["velocity"]` would be an even better thing to do (on all extension) since it ensure that we don't miss a change